node/process_manager/src/main.rs

```diff
@@ -0,0 +1,138 @@
+use chrono::prelude::Utc;
+use std::env;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::net::UdpSocket;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::thread;
+
+fn main() {
+    let start_port = 8100;
+    let end_port = 8200;
+
+    if let Some(available_port) = (start_port..end_port).find(|port| port_is_available(*port)) {
+        _ = write_log_to_file(&format!("Setting DD_DOGSTATSD_PORT to {}", available_port));
+        env::set_var("DD_DOGSTATSD_PORT", available_port.to_string());
+    } else {
+        _ = write_log_to_file(&format!(
+            "No available ports in the range {} to {}. Setting DD_DOGSTATSD_PORT the default value of 8125",
+            start_port, end_port
+        ));
+    }
```

Comment:
what do you think about extracting this logic into a pure function? this way you could add a unit tests against it as it seems that this logic is pretty isolated from the remaining of the main function

Reply:
Agreed, done!